core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServiceMethod.java

```diff
@@ -638,6 +665,34 @@ static Parameter ofRequestObject(Class<?> type,
             this.name = name;
             this.defaultValue = defaultValue;
             this.requestConverterFunction = requestConverterFunction;
+
+            if (type.isEnum()) {
+                final Set<? extends Enum<?>> enumInstances = EnumSet.allOf(type.asSubclass(Enum.class));
+                upperCaseEnum = enumInstances.stream().collect(
+                        toImmutableMap(e -> Ascii.toUpperCase(e.name()), Function.identity(), (e1, e2) -> e1));
+                if (enumInstances.size() != upperCaseEnum.size()) {
+                    upperCaseEnum = null;
+                    isCaseSensitiveEnum = true;
+                } else {
+                    isCaseSensitiveEnum = false;
+                }
+            } else {
+                upperCaseEnum = null;
+                isCaseSensitiveEnum = false;
+            }
+        }
+
+        <T extends Enum<T>> T get(String str) {
+            if (isCaseSensitiveEnum) {
+                return Enum.valueOf(type.asSubclass(Enum.class), str);
+            }
+            @SuppressWarnings("unchecked")
+            final T result = (T) upperCaseEnum.get(Ascii.toUpperCase(str));
```

Comment:
nit: If this method is called for the type which is not an enum, NPE may be thrown here. In that case, it'd be better to throw an exception like `IllegalStateException`. Also, how about changing the method name? I think `get` is too general.

Reply:
Checked & Updated Thanks